doc/chapters/api/index.rst

```diff
@@ -24,88 +13,276 @@ Methods
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
+        - **pathInHost** ``string`` path to the directory in host.
+        - **subPathInContainer** ``string`` path to the directory in container.
+        - **readOnly** ``bool`` availability of binded folder 
+
+:Return Value:
+        - **pathInContainer** ``string`` path to the bind folder in container. 
 
-string pathInContainer BindMountFolderInContainer(uint32 containerID, string pathInHost, string subPathInContainer, ool readOnly)
+
+:Example Usage:
+
+::
+
+        dbus-send --system --print-reply \
+        --dest=com.pelagicore.SoftwareContainerAgent \
+        /com/pelagicore/SoftwareContainerAgent \
+        com.pelagicore.SoftwareContainerAgent.BindMountFolderInContainer \
+        uint32:1 \
+        string:"/home/myUser/myBindFolder" \
+        string:"/home/vagrant/softwarecontainer/build" \
+        boolean:false
+        
 
 SetGatewayConfigs
 -----------------
-:containerID: uint32
-:configs: map<string, string>
+Sets the configuration of particular gateway. The gateway configuration contains settings as key/value pairs.
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
+        - **configs** ``map<string, string>`` A map to key/value pairs.
+
+:Return Value:
+       - **None**
+
+ 
+:Example Usage:
+
+::
+
+        dbus-send --system --print-reply \
+        --dest=com.pelagicore.SoftwareContainerAgent \
+        /com/pelagicore/SoftwareContainerAgent \
+        com.pelagicore.SoftwareContainerAgent.SetGatewayConfigs \
+        uint32:1 \
+        dict:string:string:"configItem1","1","configItem2","2"
 
-void SetGatewayConfigs(uint32 containerID, map<string, string> configs)
 
 SetCapabilities
 ---------------
-:containerID: uint32
-:capabilities: array<string>
+Currently This method has no applicable usage. 
+
+:Parameters:
+        - **containerID** ``uint32`` The ID obtained by CreateContainer method.
+        - **capabilities** ``array<string>``
+
+
+:Return Value:
+       - **success** ``boolean`` either true or false.
+
+:Example Usage:
+        - Althought there is no meaning to use this method, it can be used as following example.
+
+::
+
+         dbus-send --system --print-reply \
+         --dest=com.pelagicore.SoftwareContainerAgent \
+         /com/pelagicore/SoftwareContainerAgent \
+         com.pelagicore.SoftwareContainerAgent.SetCapabilities \
+         uint32:1 \
+         array:string:"I","Can","not","provide","functionality"
 
-bool success SetCapabilities(uint32 containerID, array<string> capabilities)
 
 Signals
 -------
 
 ProcessStateChanged
 -------------------
-:containerID: uint32
-:processID: uint32
-:isRunning: bool
-:exitCode: uint32
+The Dbus API sends signal when process state is changed. There are four value to be emitted.
+
+:containerID: ``uint32`` The ID obtained by CreateContainer method.
```

Comment:
A different formatting standard than in the previous D-Bus calls is used here. Same formatting all over would be nice.